README.md

```diff
@@ -1 +1,247 @@
+For affected devices, the following statement in section 38.7.4 of the reference
+manual only holds when the FIFO is not empty:
+
+    "When a new value is written, the duty cycle changes after the current
+     period is over."
+
+To workaround this issue, the following two points must be met:
+1. When the FIFO is empty, the first sample added to the FIFO must be the
+   current duty cycle value from the PSWSAR.
+2. During playback, the rate at which the FIFO empties must match the rate at
+   which new samples are added to the FIFO. This will prevent the FIFO from
+   emptying or overflowing, which would respectively cause glitches or lost
+   samples.
+### Workaround
+This module implements point 1 above in the SDMA script. In order to meet point
+2, the following expression must be true:
+
+    sample_rate * pwm_period * (pwm_prescaler + 1) << pwm_repeat == ipg_clk
+
+Where `ipg_clk` is the PWM clock frequency. A warning is generated during device
+probing or `PWM_LED_CONFIGURE` if this condition is not met. For devices not
+affected by this issue, this warning can be safely ignored.
+
+## Device Tree Bindings
+```
+Required properties:
+- compatible : Should be "unu,imx-pwm-led"
+- unu,sdma-script-origin : **Only required for the first device**
+  Location to load the SDMA script (in 32-bit data words).
+  The 4KB SDMA internal RAM area for scripts and data is located at
+  0xc00 - 0xfff. The imx-pwm-led script is 10 words long.
+- unu,timer : **Only required for the first device** The EPIT timer used.
+- unu,sdma-channel : The SDMA channel number.
+
+Example:
+&epit1 {
+	compatible = "fsl,imx6ul-epit";
+	id = <0>;
+	clocks = <&clks IMX6UL_CLK_EPIT1>;
+	clock-names = "epit1";
+	sdma-event = <16>;
+};
+
+&pwm1 {
+	compatible = "unu,imx-pwm-led";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pwm1>;
+	status = "okay";
+	unu,timer = <&epit1>;
+	unu,sdma-script-origin = <0xc00>;
+	unu,sdma-channel = <24>;
+};
+
+&pwm2 {
+	compatible = "unu,imx-pwm-led";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pwm2>;
+	status = "okay";
+	unu,sdma-channel = <25>;
+};
+```
+
+## Building
+Building is normally performed using [Yocto](https://www.yoctoproject.org/).
+
+Here is an example `bitbake` script:
+```
+SUMMARY = "Kernel loadable module for i.MX PWM LEDs"
+LICENSE = "GPLv2"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=b234ee4d69f5fce4486a80fdaf4a4263"
+
+inherit module
+
+SRCBRANCH = "master"
+SRC_GIT = "git://git@github.com/unumotors/kernel-module-imx-pwm-led.git;protocol=ssh"
+SRC_URI = "${SRC_GIT};branch=${SRCBRANCH}"
+SRCREV = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
+
+S = "${WORKDIR}/git"
+
+# The inherit of module.bbclass will automatically name module packages with
+# "kernel-module-" prefix as required by the oe-core build environment.
+
+RPROVIDES_${PN} += "kernel-module-imx-pwm-led"
+
+do_install_append() {
+    install -d ${D}${includedir}/linux
+    install -m 0644 ${S}/pwm_led.h ${D}${includedir}/linux
```

Comment:
Since this isn't a portable module, the include directory should probably be `<imx/linux/pwm_led.h>` rather than `<linux/pwm.h>`

Reply:
Fixed